xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-test/xwiki-platform-flamingo-skin-test-docker/src/test/it/org/xwiki/flamingo/test/docker/LoginIT.java

```diff
@@ -164,6 +162,8 @@ void correctUrlIsAccessedAfterLogin(TestUtils setup, TestReference testReference
         // Navigate to it and verify we're asked to log in and that we're not logged-in.
         setup.gotoPage(testReference);
         LoginPage loginPage = new LoginPage();
+        // Waits for the page the redirection before asserting the location.
+        setup.getDriver().waitUntilPageIsReloaded();
```

Comment:
It's not good you need to add a reload marker first, else this wait is not taken into account. Check the documentation of `waitUntilPageIsReloaded`.

Reply:
`addPageNotYetReloadedMarker` added above.

Reply:
Also (I know I keep repeating this but it's important), don't explain what the code does, that's not very useful. Instead explain why you do it. In this case does it mean there's some JS executed in between? If you wan't write a proper explanation to why then it means you don't understand what you're doing! At least mention that so that others will not think you knew what you were doing ;) But best is to not put stuff you don't understand and thus always explain the why.